wifi.go

```diff
@@ -0,0 +1,134 @@
+package keylight
+
+import (
+	"bytes"
+	"context"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+)
+
+// A WiFiInfo contains the information neccessary for a Key Light to connect
+// to a wireless network.
+type WiFiInfo struct {
+	SSID         string `json:"ssid,omitempty"`
+	Passphrase   string `json:"passphrase,omitempty"`
+	SecurityType int    `json:"securityType,omitempty"`
+}
+
+const (
+	SecurityTypeNone int = 0
```

Comment:
Let's make these a new enum-like type and possibly consider using stringer to generate strings. I have also started removing type prefixes when the API is pretty small:

```go
type WiFiSecurity int

const (
    None WiFiSecurity = 0
    WEP WiFiSecurity = 1
    PSK WiFiSecurity = 2
)
```

Interestingly the Web UI doesn't seem to allow WEP (probably for the best) and refers to PSK as WPA/WPA2 Personal. Thoughts on renaming the latter?